Sources/SwiftSyntaxMacroExpansion/MacroReplacement.swift

```diff
@@ -156,6 +177,41 @@ fileprivate class ParameterReplacementVisitor: SyntaxAnyVisitor {
     return .visitChildren
   }
 
+  override func visit(_ node: GenericArgumentClauseSyntax) -> SyntaxVisitorContinueKind {
+    return .visitChildren
+  }
+
```

Comment:
IIUC you only want to visit the `GenericArgumentClauseSyntax` of the `MacroExpansion{Expr, Decl}Syntax`, right? I think we should check that the parent is that type. Otherwise you also walk into the `GenericArgumentClauseSyntax` of eg. a type. 

A test case I would expect to hit this is

```swift
macro gen(a: Array<Int>) = …
```

We shouldn’t walk the `GenericArgumentClauseSyntax` of `Array`.

Reply:
Oh thank you, I'll handle that

Reply:
AFAICT we don’t handle this yet. Or am I missing something?

Reply:
I did check the snippet and worked fine, I'll add more tests but I guess I'm not sure how to cause a bad case?

Reply:
Ah, yes. Seems to work. I didn’t realize we were only walking the definition and that it can’t contain complex types because it can only contain literals.

Reply:
Cool, glad I didn't misunderstand then :)